client/src/components/License/LicenseSelector.vue

```diff
@@ -1,130 +1,111 @@
+<script setup lang="ts">
+import { BAlert } from "bootstrap-vue";
+import { computed, ref, watch } from "vue";
+import Multiselect from "vue-multiselect";
+
+import { GalaxyApi } from "@/api";
+import { type components } from "@/api/schema";
+import { errorMessageAsString } from "@/utils/simple-error";
+
+import License from "@/components/License/License.vue";
+import LoadingSpan from "@/components/LoadingSpan.vue";
+
+const defaultLicense: LicenseType = {
+    licenseId: null,
+    name: "*Do not specify a license.*",
+};
+
+type LicenseMetadataModel = components["schemas"]["LicenseMetadataModel"];
+type LicenseType = {
+    licenseId: string | null;
+    name: string;
+};
+
+interface Props {
+    inputLicense: string;
+}
+
+const props = defineProps<Props>();
+
+const emit = defineEmits<{
+    (e: "onLicense", license: string | null): void;
+}>();
+
+const licensesLoading = ref(true);
+const errorMessage = ref<string>("");
+const currentLicense = ref<LicenseType>();
+const licenses = ref<LicenseMetadataModel[] | undefined>([]);
+
+const licenseOptions = computed(() => {
+    const options: LicenseType[] = [];
+
+    options.push(defaultLicense);
+
+    for (const license of licenses.value || []) {
+        if (license.licenseId == currentLicense.value?.licenseId || license.recommended) {
+            options.push({
+                licenseId: license.licenseId,
+                name: license.name,
+            });
+        }
+    }
+
+    return options;
+});
+
+function onLicense(license: LicenseType) {
+    emit("onLicense", license.licenseId);
+}
+
+async function fetchLicenses() {
+    const { error, data } = await GalaxyApi().GET("/api/licenses");
+
+    if (error) {
+        errorMessage.value = errorMessageAsString(error) || "Unable to fetch licenses.";
+    }
+
+    licenses.value = data;
```

Comment:
Could we put this in a store, so it doesn't reload on re-rendering this component?